net/mwan3/files/usr/sbin/mwan3

```diff
@@ -37,52 +38,64 @@ ifdown()
 
 	ACTION=ifdown INTERFACE=$1 /sbin/hotplug-call iface
 
-	kill $(pgrep -f "mwan3track $1 $2") &> /dev/null
+	kill $(pgrep -f "mwan3track $1 ") &> /dev/null
 	mwan3_track_clean $1
 }
 
 ifup()
 {
-	local device enabled up l3_device status
-
-	mwan3_lock "command" "mwan3"
-
-	config_load mwan3
-	config_get_bool enabled globals 'enabled' 0
-
-	[ ${enabled} -gt 0 ] || {
-		echo "The service mwan3 is global disabled."
-		echo "Please execute \"/etc/init.d/mwan3 start\" first."
-		mwan3_unlock "command" "mwan3"
-		exit 1
-	}
+	local device enabled up l3_device status interface true_iface
 
 	if [ -z "$1" ]; then
 		echo "Expecting interface. Usage: mwan3 ifup <interface>"
-		mwan3_unlock "command" "mwan3"
 		exit 0
 	fi
 
 	if [ -n "$2" ]; then
 		echo "Too many arguments. Usage: mwan3 ifup <interface>"
-		mwan3_unlock "command" "mwan3"
 		exit 0
 	fi
 
-	config_get enabled "$1" enabled 0
-	mwan3_unlock "command" "mwan3"
+	interface=$1
+
+	if [ "${MWAN3_STARTUP}" != 1 ]; then
+		# It is not necessary to obtain a lock here, because it is obtained in the hotplug
+		# script, but we still want to do the check to print a useful error message
+		config_load mwan3
+		config_get_bool enabled globals 'enabled' 0
+
+		[ ${enabled} -gt 0 ] || {
+			echo "The service mwan3 is global disabled."
+			echo "Please execute \"/etc/init.d/mwan3 start\" first."
+			exit 1
+		}
+	else
+		enabled=1
+	fi
+	mwan3_get_true_iface true_iface $interface
+	status=$(ubus -S call network.interface.$true_iface status)
 
-	status=$(ubus -S call network.interface.$1 status)
 	[ -n "$status" ] && {
 		json_load "$status"
 		json_get_vars up l3_device
 	}
+	hotplug_startup()
+	{
+		MWAN3_STARTUP=$MWAN3_STARTUP ACTION=ifup INTERFACE=$INTERFACE DEVICE=$l3_device TRUE_INTERFACE=$true_iface sh /etc/hotplug.d/iface/15-mwan3
+		MWAN3_STARTUP=$MWAN3_STARTUP ACTION=ifup INTERFACE=$INTERFACE DEVICE=$l3_device TRUE_INTERFACE=$true_iface sh /etc/hotplug.d/iface/16-mwan3-user
```

Comment:
should be `INTERFACE=$interface` ?

interface is local var and set by `interface=$1` in line 59

Reply:
yes. ty